### PR TITLE
[kube-prometheus-stack] Thanos Ruler: Support omit externalPrefix with externalPrefixNilUsesHelmValues=true, if ingress is enabled

### PR DESCRIPTION
**Pull request by jkroepke**
#### What this PR does / why we need it

I would like to use `web.prefix-header` instead an static prefix from `externalPrefix`. Thats works only, if externalPrefix is fully omit.

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by jkroepke** [comment]
@QuentinBisson @GMartinez-Sisti 


If you could take an eye here, I would appreciate it.

**Review by QuentinBisson — APPROVED** [review]
LGTM, can you bump the version?

**Comment by jkroepke** [comment]
> LGTM, can you bump the version?

Sure, I have to.

File: charts/kube-prometheus-stack/templates/thanos-ruler/ruler.yaml [hunk 1]
```diff
@@ -31,10 +31,12 @@ spec:
   listenLocal: {{ .Values.thanosRuler.thanosRulerSpec.listenLocal }}
   serviceAccountName: {{ template "kube-prometheus-stack.thanosRuler.serviceAccountName" . }}
 {{- if .Values.thanosRuler.thanosRulerSpec.externalPrefix }}
-  externalPrefix: "{{ tpl .Values.thanosRuler.thanosRulerSpec.externalPrefix . }}"
+  externalPrefix: {{ tpl .Values.thanosRuler.thanosRulerSpec.externalPrefix . | quote }}
```
Comment (QuentinBisson) [inline charts/kube-prometheus-stack/templates/thanos-ruler/ruler.yaml:34]: @jkroepke coming back to reviewing those, is that still needed with the PR to clean up the `UsesHelmValues ` ?

**Pull request closed without merging**